ilastik/applets/dataSelection/dataSelectionGui.py

```diff
@@ -571,13 +572,37 @@ def _createDatasetInfo(self, roleIndex: int, filePath: Path, roi=None) -> Filesy
             self._add_default_inner_path(roleIndex=roleIndex, inner_path=selected_dataset)
             data_path = data_path / re.sub("^/", "", selected_dataset)
 
-        return RelativeFilesystemDatasetInfo.create_or_fallback_to_absolute(
+        return self.instantiate_dataset_info(
+            role=roleIndex,
+            info_creator=RelativeFilesystemDatasetInfo.create_or_fallback_to_absolute,
             filePath=data_path.as_posix(),
             project_file=self.project_file,
             allowLabels=(self.guiMode == GuiMode.Normal),
             subvolume_roi=roi,
         )
 
+    def guess_axistags_for(self, role: Union[str, int], info: DatasetInfo) -> Optional[AxisTags]:
+        if self.topLevelOperator.num_lanes == 0:
+            return info.axistags
+        lane = self.topLevelOperator.getLane(-1)
+        previous_info = lane.get_dataset_info(role)
+        if previous_info is None or previous_info.default_tags == previous_info.axistags:
+            return info.axistags
+        if previous_info.__class__ != info.__class__ or len(previous_info.axiskeys) != len(info.axiskeys):
```

Comment:
I would argue this comparison is too strong:

* one could have copied the last data to project internal -> won't work anymore
* even if one dataset can be resolved to `RelativeDatasetInfo` and the other one "only" to `AbsoluteDatasetInfo` it won't work.

There are arguments for and against that. I don't know what happens more often in reality. But as it stands right now, having two identical files in different folders could result in the axistags being properly applied in one case, and not in the other.

Reply:
Fair point. I'll remove or refine the class comparison